README.md
# Clean Times

[![codecov](https://img.shields.io/codecov/c/github/josegoval/clean-times?style=for-the-badge)](https://codecov.io/gh/josegoval/clean-times)
![npm bundle size](https://img.shields.io/bundlephobia/min/clean-times?style=for-the-badge)
[![https://nodei.co/npm/clean-times.png?downloads=true&downloadRank=true&stars=true](https://nodei.co/npm/clean-times.png?downloads=true&downloadRank=true&stars=true)](https://www.npmjs.com/package/clean-times)

`clean-times` is a JS and TS library that help you write clean code times.

**Feel like supporting this free plugin?**

<a href="https://www.buymeacoffee.com/josegoval" target="_blank"><img src="https://cdn.buymeacoffee.com/buttons/v2/default-yellow.png" alt="Buy Me A Coffee" style="height: 60px !important;width: 217px !important;" ></a>

**Table of Contents**

- [Installation](#installation)
- [Example usage](#example-usage)
- [Main Functions](#main-functions)
- [Utility functions](#utility-functions)
- [Other utilities](#other-utilities)

## Installation

```bash
npm install --save clean-times
```

or

```bash
yarn add clean-times
```

## Example usage

Import and use as follow with ES6 syntax:

```ts
import { getDaysInMS, FIVE_DAYS_IN_MS } from "clean-times";

console.log(getDaysInMs(5)) 
console.log(FIVE_DAYS_IN_MS) 
```

For node environments without ES6 use `require` as follows:

```ts

const { getDaysInMS, FIVE_DAYS_IN_MS } = require("clean-times")

console.log(getDaysInMs(5)) 
console.log(FIVE_DAYS_IN_MS) 
```

CHANGELOG.md
# 1.0.0 (2023-09-09)


### Features

* core package setup ([fb00e4e](https://github.com/josegoval/clean-times/commit/fb00e4e78df4ecd48bb1256009fc2e47724e517e))

# 1.0.0 (2023-09-09)


### Features

* initial version ([00f54c1](https://github.com/josegoval/clean-times/commit/00f54c16651c9f413effa2250c2578d60cbca639))
* millisecond constants and utils ([e179d1d](https://github.com/josegoval/clean-times/commit/e179d1d9c49862c748cfab885f64f1ff13a719ee))
* milliseconds converters and constants ([2a78029](https://github.com/josegoval/clean-times/commit/2a78029348989986a83f5be800ecb4668157fd51))
* seconds converters and constants ([d32a352](https://github.com/josegoval/clean-times/commit/d32a352997f5e9a352a605ea6cf712cda9912194))
* squashing changes ([c00cdd1](https://github.com/josegoval/clean-times/commit/c00cdd12b8024f7f60a5f95f7c811a06b1183c44))


src/times/seconds/constants.test.ts
import { A_DAY_IN_SECONDS, A_MINUTE_IN_SECONDS, A_WEEK_IN_SECONDS, A_YEAR_IN_SECONDS, EIGHT_DAYS_IN_SECONDS, EIGHT_MINUTES_IN_SECONDS, EIGHT_YEARS_IN_SECONDS, FIFTEEN_MINUTES_IN_SECONDS, FIVE_DAYS_IN_SECONDS, FIVE_MINUTES_IN_SECONDS, FIVE_YEARS_IN_SECONDS, FORTY_FIVE_MINUTES_IN_SECONDS, FOUR_DAYS_IN_SECONDS, FOUR_MINUTES_IN_SECONDS, FOUR_YEARS_IN_SECONDS, NINE_DAYS_IN_SECONDS, NINE_MINUTES_IN_SECONDS, NINE_YEARS_IN_SECONDS, SEVEN_DAYS_IN_SECONDS, SEVEN_MINUTES_IN_SECONDS, SEVEN_YEARS_IN_SECONDS, SIX_DAYS_IN_SECONDS, SIX_MINUTES_IN_SECONDS, SIX_YEARS_IN_SECONDS, TEN_DAYS_IN_SECONDS, TEN_MINUTES_IN_SECONDS, TEN_YEARS_IN_SECONDS, THIRTY_MINUTES_IN_SECONDS, THREE_DAYS_IN_SECONDS, THREE_MINUTES_IN_SECONDS, THREE_WEEK_IN_SECONDS, THREE_YEARS_IN_SECONDS, TWO_DAYS_IN_SECONDS, TWO_MINUTES_IN_SECONDS, TWO_WEEK_IN_SECONDS, TWO_YEARS_IN_SECONDS } from "./constants";
import { getDaysInSeconds, getHoursInSeconds } from "./converters";

describe("Seconds module", () => {
	it("converts minutes", ()=> {
		expect(A_MINUTE_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS);
		expect(TWO_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 2);
		expect(THREE_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 3);
		expect(FOUR_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 4);
		expect(FIVE_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 5);
		expect(SIX_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 6);
		expect(SEVEN_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 7);
		expect(EIGHT_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 8);
		expect(NINE_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 9);
		expect(TEN_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 10);
		expect(FIFTEEN_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 15);
		expect(THIRTY_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 30);
		expect(FORTY_FIVE_MINUTES_IN_SECONDS).toBe(A_MINUTE_IN_SECONDS * 45);
	});

	it("converts days", ()=> {
		const aDayInSeconds = getHoursInSeconds(24);
		expect(A_DAY_IN_SECONDS).toBe(aDayInSeconds);
		expect(TWO_DAYS_IN_SECONDS).toBe(aDayInSeconds * 2);
		expect(THREE_DAYS_IN_SECONDS).toBe(aDayInSeconds * 3);
		expect(FOUR_DAYS_IN_SECONDS).toBe(aDayInSeconds * 4);
		expect(FIVE_DAYS_IN_SECONDS).toBe(aDayInSeconds * 5);
		expect(SIX_DAYS_IN_SECONDS).toBe(aDayInSeconds * 6);
		expect(SEVEN_DAYS_IN_SECONDS).toBe(aDayInSeconds * 7);
		expect(EIGHT_DAYS_IN_SECONDS).toBe(aDayInSeconds * 8);
		expect(NINE_DAYS_IN_SECONDS).toBe(aDayInSeconds * 9);
		expect(TEN_DAYS_IN_SECONDS).toBe(aDayInSeconds * 10);
	});

	it("converts weeks", ()=> {
		expect(A_WEEK_IN_SECONDS).toBe(SEVEN_DAYS_IN_SECONDS);
		expect(TWO_WEEK_IN_SECONDS).toBe(SEVEN_DAYS_IN_SECONDS * 2);
		expect(THREE_WEEK_IN_SECONDS).toBe(SEVEN_DAYS_IN_SECONDS * 3);
	});

	it("converts years", ()=> {
		const aYearInSeconds = getDaysInSeconds(365);
		expect(A_YEAR_IN_SECONDS).toBe(aYearInSeconds);
		expect(TWO_YEARS_IN_SECONDS).toBe(aYearInSeconds * 2);
		expect(THREE_YEARS_IN_SECONDS).toBe(aYearInSeconds * 3);
		expect(FOUR_YEARS_IN_SECONDS).toBe(aYearInSeconds * 4);
		expect(FIVE_YEARS_IN_SECONDS).toBe(aYearInSeconds * 5);
		expect(SIX_YEARS_IN_SECONDS).toBe(aYearInSeconds * 6);
		expect(SEVEN_YEARS_IN_SECONDS).toBe(aYearInSeconds * 7);
		expect(EIGHT_YEARS_IN_SECONDS).toBe(aYearInSeconds * 8);
		expect(NINE_YEARS_IN_SECONDS).toBe(aYearInSeconds * 9);
		expect(TEN_YEARS_IN_SECONDS).toBe(aYearInSeconds * 10);
	});
});

src/times/seconds/converters.test.ts
import { getDaysInSeconds, getYearsInSeconds } from "./converters";

describe("Converters module", () => {
	it("converts leap years", ()=> {
		expect(getYearsInSeconds(3, { isLeapYear: true })).toBe(getDaysInSeconds(366) * 3);
	});

});

src/times/seconds/constants.ts
import { getDaysInSeconds, getHoursInSeconds, getMinutesInSeconds, getWeeksInSeconds, getYearsInSeconds } from "./converters";

export const A_MINUTE_IN_SECONDS = getMinutesInSeconds(1);
export const TWO_MINUTES_IN_SECONDS = getMinutesInSeconds(2);
export const THREE_MINUTES_IN_SECONDS = getMinutesInSeconds(3);
export const FOUR_MINUTES_IN_SECONDS = getMinutesInSeconds(4);
export const FIVE_MINUTES_IN_SECONDS = getMinutesInSeconds(5);
export const SIX_MINUTES_IN_SECONDS = getMinutesInSeconds(6);
export const SEVEN_MINUTES_IN_SECONDS = getMinutesInSeconds(7);
export const EIGHT_MINUTES_IN_SECONDS = getMinutesInSeconds(8);
export const NINE_MINUTES_IN_SECONDS = getMinutesInSeconds(9);
export const TEN_MINUTES_IN_SECONDS = getMinutesInSeconds(10);
export const FIFTEEN_MINUTES_IN_SECONDS = getMinutesInSeconds(15);
export const THIRTY_MINUTES_IN_SECONDS = getMinutesInSeconds(30);
export const FORTY_FIVE_MINUTES_IN_SECONDS = getMinutesInSeconds(45);

export const A_HOUR_IN_SECONDS = getHoursInSeconds(1);
export const TWO_HOURS_IN_SECONDS = getHoursInSeconds(2);
export const THREE_HOURS_IN_SECONDS = getHoursInSeconds(3);
export const FOUR_HOURS_IN_SECONDS = getHoursInSeconds(4);
export const FIVE_HOURS_IN_SECONDS = getHoursInSeconds(5);
export const SIX_HOURS_IN_SECONDS = getHoursInSeconds(6);
export const SEVEN_HOURS_IN_SECONDS = getHoursInSeconds(7);
export const EIGHT_HOURS_IN_SECONDS = getHoursInSeconds(8);
export const NINE_HOURS_IN_SECONDS = getHoursInSeconds(9);
export const TEN_HOURS_IN_SECONDS = getHoursInSeconds(10);

export const A_DAY_IN_SECONDS = getDaysInSeconds(1);
export const TWO_DAYS_IN_SECONDS = getDaysInSeconds(2);
export const THREE_DAYS_IN_SECONDS = getDaysInSeconds(3);
export const FOUR_DAYS_IN_SECONDS = getDaysInSeconds(4);
export const FIVE_DAYS_IN_SECONDS = getDaysInSeconds(5);
export const SIX_DAYS_IN_SECONDS = getDaysInSeconds(6);
export const SEVEN_DAYS_IN_SECONDS = getDaysInSeconds(7);
export const EIGHT_DAYS_IN_SECONDS = getDaysInSeconds(8);
export const NINE_DAYS_IN_SECONDS = getDaysInSeconds(9);
export const TEN_DAYS_IN_SECONDS = getDaysInSeconds(10);

export const A_WEEK_IN_SECONDS = getWeeksInSeconds(1);
export const TWO_WEEK_IN_SECONDS = getWeeksInSeconds(2);
export const THREE_WEEK_IN_SECONDS = getWeeksInSeconds(3);

export const A_YEAR_IN_SECONDS = getYearsInSeconds(1);
export const TWO_YEARS_IN_SECONDS = getYearsInSeconds(2);
export const THREE_YEARS_IN_SECONDS = getYearsInSeconds(3);
export const FOUR_YEARS_IN_SECONDS = getYearsInSeconds(4);
export const FIVE_YEARS_IN_SECONDS = getYearsInSeconds(5);
export const SIX_YEARS_IN_SECONDS = getYearsInSeconds(6);
export const SEVEN_YEARS_IN_SECONDS = getYearsInSeconds(7);
export const EIGHT_YEARS_IN_SECONDS = getYearsInSeconds(8);
export const NINE_YEARS_IN_SECONDS = getYearsInSeconds(9);
export const TEN_YEARS_IN_SECONDS = getYearsInSeconds(10);
